### PR TITLE
chore: upgrade GitHub CI actions

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -15,11 +15,12 @@ jobs:
...
       - name: Check out the master branch
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Set up JDK 1.11
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v2
         with:
-          java-version: 1.11
+          distribution: 'temurin'
+          java-version: '19'
```
Comment (mlutze) [inline .github/workflows/build.yaml:23]: We no longer support Java 11?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -15,11 +15,12 @@ jobs:
 
     steps:
       - name: Check out the master branch
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Set up JDK 1.11
```
Comment (mlutze) [inline .github/workflows/build.yaml:19]: Should update to match (or remove number)

**Comment by magnus-madsen** [comment]
Still working on this PR.

The plan was always to migrate to whatever LTS arrived with support for VirtualThreads.

I am slowing moving in that direction.

**Comment by magnus-madsen** [comment]
@mlutze Request approval :)

**Review by mlutze — APPROVED** [review]

**Comment by magnus-madsen** [comment]
!!!jar

**Pull request merged**